test/Microsoft.AspNet.Mvc.Core.Test/Filters/FormatFilterTest.cs

```diff
@@ -0,0 +1,253 @@
+            var filter = new FormatFilter();
+
+            // Act
+            filter.OnResourceExecuting(context);
+
+            // Assert
+            var result = context.Result as IActionResult;
+            Assert.False(result is HttpNotFoundResult);
+        }
+
+        [Theory]
+        [InlineData("json", FormatPlace.RouteData, "application/json")]
+        [InlineData("json", FormatPlace.QueryData, "application/json")]
+        public void FormatFilter_ContextContainsFormat_ContainsProducesFilter_Conflicting(
+            string format,
+            FormatPlace place,
+            string contentType)
+        {
+            // Arrange
+            var mediaType = MediaTypeHeaderValue.Parse(contentType);
+            var produces = new ProducesAttribute("application/xml", new string[] { "application/foo", "text/bar" });
+            var context = CreateResourceExecutingContext(new IFilter[] { produces }, format, place);
+            var filter = new FormatFilter();
+
+            // Act
+            filter.OnResourceExecuting(context);
+
+            // Assert
+            var result = context.Result as IActionResult;
+            Assert.True(result is HttpNotFoundResult);
+        }
+
+        private static ResourceExecutingContext CreateResourceExecutingContext(
+            IFilter[] filters,
+            string format = null, 
+            FormatPlace? place = null)
+        {
+            if(format == null || place == null)
+            {
+                var context = new ResourceExecutingContext(
+                    CreateActionContext(),
+                    filters);
+                context.Result = new HttpStatusCodeResult(200);
+                return context;
+            }
+
+            var context1 = new ResourceExecutingContext(
+                CreateActionContext(format, place),
+                filters);
+            context1.Result = new HttpStatusCodeResult(200);
+            return context1;
+        }
+
+        private static ResultExecutingContext CreateResultExecutingContext(
+            string format = null,
+            FormatPlace? place = null)
+        {
+            if (format == null || place == null)
+            {
+                return new ResultExecutingContext(
+                    new ActionContext(new DefaultHttpContext(), new RouteData(), new ActionDescriptor()),
+                    new IFilter[] { },
+                    new ObjectResult("Some Value"));
+            }
+
+            return new ResultExecutingContext(
+                CreateActionContext(format, place),
+                new IFilter[] { },
+                new ObjectResult("Some Value"));
+        }
+
+        private static ActionContext CreateActionContext(string format = null, FormatPlace? place = null)
+        {
+            var httpContext = CreateMockHttpContext();
+
+            if (place == FormatPlace.RouteData || place == FormatPlace.RouteAndQueryData)
```

Comment:
Move these setups to individual tests. Having conditional setups that are dependent on test data removed from the actual test code makes it harder to follow what's happening